content/blog/build-your-own-github-service.md

```diff
@@ -0,0 +1,276 @@
+| 📂 `remote.git` | A bare repository (The one where all changes are pushed).                |
+
+> **Recommended**: It's a good practice to name your bare repository with a `.git` suffix, (like `remote.git`).
+
+> When you see any github repository clonable link (whether in HTTP or SSH URL) that always ends with `.git`, it indicates that **on the server, the bare repository is created in a folder named exactly as `repository_name.git`**.
+
+**2. Set Up the Bare Repository**
+
+Now, go into the remote folder and create the bare repository
+
+```bash
+# Navigate to remote.git directory
+cd remote.git
+
+# Create bare repository here
+git init --bare
+```
+
+**3. Clone the Bare Repository**
+
+Go back to your project directory and navigate to the `local` folder. Clone the bare repository here
+
+```bash
+# Navigate to your working directory where you want to clone
+cd local
+
+# clone the bare repository
+git clone <path-to-bare-repo>
+# Replace <path-to-bare-repo> with the actual file path or URL of your bare repository.
+# For an example: C:\User\DELL\Experiment\remote.git\
+```
+
+You should now see a folder named `remote` inside your `local` folder, as our bare repository was named `remote.git`.
+
+> **Note**: This is not the bare repository's directory. It’s now your local repository, created by cloning the bare repository.
+
+**4. Make Some Changes in the Local (Clonned) Repository**
+
+Now, inside your local repository, navigate to the `remote` folder and do the following:
+
+```bash
+# Navigate to remote
+cd remote
+
+# Create README.md file
+echo "README" > README.md
+
+# Add to staging area
+git add .
+
+# Commit the changes
+git commit -m "Add: README.md"
+```
+
+**5. Check the Remote Origin**
+
+To see where your remote repository is set up, run:
+
+```bash
+git remote -vv
+```
+
+You should see the remote URL listed under "`origin`" :
+
+For an example:
+
+```bash
+origin C:\User\DELL\Experiment\remote.git\ (fetch)
+origin C:\User\DELL\Experiment\remote.git\ (push)
+```
+
+If **the remote isn't set up yet, you can add** it by running this command:
+
+```bash
+git remote add origin "<path-to-bare-repo>"
+# Replace <path-to-bare-repo> with the actual file path or URL of your bare repository.
+# For an example: C:\User\DELL\Experiment\remote.git\
+```
+
+**6. Push changes to the Bare Repository**
+
+Now that the remote repository url is set up, push your changes to the bare repository:
+
+```bash
+git push
+```
+
+**7. Verify the Bare Repository by Cloning It Elsewhere**
+
+Finally, to make sure everything works, let’s try cloning the bare repository in a different folder (e.g., bare_local). you should able to see changes.
+
+**This is how you can create a local Git repository, connect it to a bare repository, and push changes.**
+
+<br>
+
+# <p align="center">...</p>
+
+<br>
+
+<h4>2️⃣ Experiment: Hosting a Bare Git Repository on a Remote Server</h4>
```

Comment:
This section can use some abstraction, instead of making it aws specific, you can present it as just a computer on the cloud

- Have a server ready with ssh access
- SSH in and and run the commands
- Clone the repo and use it